### PR TITLE
Port to Xcode 7/Swift 2.0

### LINKED ISSUE #2062: Support Swift 2
I know that Swift 2 has just been published but for future compatibility it will be great to fix it.

I'm using cocoa pod with Realm (0.93.1) RealmSwift (0.93.1)

screen shot 2015-06-09 at 10 30 38

### PR DESCRIPTION
**Pull request by tgoyne**
Closes #2062.

To use this with CocoaPods:
```ruby
# Podfile
pod 'Realm', :git => 'https://github.com/realm/realm-cocoa', :branch => 'swift-2.0'

# Uncomment the following line if you're using RealmSwift
# Note that the Realm pod must be explicitly listed as above even if you're using RealmSwift
# pod 'RealmSwift', :git => 'https://github.com/realm/realm-cocoa', :branch => 'swift-2.0'
```

If you aren't using CocoaPods, clone the repository, checkout the swift-2.0 branch, and then run `sh build.sh ios-static`, `sh build.sh ios-dynamic`, `sh build.sh ios-swift`, `sh build.sh osx`, or `sh build.sh osx-swift` depending on which framework you need. The built framework will be placed in the `build` directory.

### CODE REVIEW

File: RealmSwift/Tests/RealmTests.swift [hunk 1]
```diff
@@ -31,80 +31,79 @@ class RealmTests: TestCase {
...
-        XCTAssertNil(Realm(path: Realm.defaultPath, readOnly: false, error: &error), "Should not throw with error")
-        XCTAssertNotNil(error)
+        do {
+            try Realm(path: Realm.defaultPath, readOnly: false)
+            XCTFail("Realm creation should have failed")
+        } catch {
+        }
 
-        assertThrows(Realm(path: Realm.defaultPath, readOnly: false, error: nil))
-        assertThrows(Realm(path: Realm.defaultPath, readOnly: false))
```
Comment (jpsim) [inline RealmSwift/Tests/RealmTests.swift]: I think we should have kept this test case

Comment (tgoyne) [inline RealmSwift/Tests/RealmTests.swift]: The scenario being tested there does not have an equivalent with swift exceptions.

File: RealmSwift/Tests/ResultsTests.swift [hunk 2]
```diff
@@ -186,7 +186,7 @@ class ResultsTests: TestCase {
         innerArray.array.append(SwiftObject())
         outerArray.array.append(innerArray)
         realm.add(outerArray)
-        XCTAssertEqual(Int(1), outerArray.array.filter("ANY array IN %@", realm.objects(SwiftObject)).count)
+        // FIXME: XCTAssertEqual(Int(1), outerArray.array.filter("ANY array IN %@", realm.objects(SwiftObject)).count)
```
Comment (jpsim) [inline RealmSwift/Tests/ResultsTests.swift]: This test still fails, so @alazier's CVarArgType fix didn't work :disappointed:.

I'd rather not have to force users to do `realm.objects(SwiftObject).map { $0 }`...

Comment (tgoyne) [inline RealmSwift/Tests/ResultsTests.swift]: It passes for me. Did you delete the `build` directory first? The RealmSwift project never rebuilds Realm.framework if it already exists, and using a build from master will compile and fail a few tests.

Comment (tgoyne) [inline RealmSwift/Tests/ResultsTests.swift]: Oh, you mean the commented-out one.

**Comment by imran20487** [comment]
From where can I get realm.framework and realmswift.framework. for xcode 7 swift 2.0?

**Comment by jpsim** [comment]
@imran20487 as you can see from this PR, we're still working on supporting Swift 2. As you're now subscribed to this issue, you'll be notified when we make progress and have anything further to share.

Thanks for your patience.

**Comment by jpsim** [comment]
@imran20487 I was mistaken, @tgoyne actually fixed the last remaining issues with getting Realm Swift to work with the Swift 2 beta last Friday.

If you'd like to try out, you can use CocoaPods with the following `Podfile` contents:

```ruby
use_frameworks!
pod 'Realm', :git => 'git@github.com:realm/realm-cocoa.git', :branch => 'swift-2.0'
pod 'RealmSwift', :git => 'git@github.com:realm/realm-cocoa.git', :branch => 'swift-2.0'
```

**Comment by anlaital** [comment]
@jpsim I'm getting the following error when trying to install the branch with CocoaPods:

```
Pre-downloading: `RealmSwift` from `git@github.com:realm/realm-cocoa.git`, branch `swift-2.0`
[!] Error installing RealmSwift
[!] /usr/bin/git clone git@github.com:realm/realm-cocoa.git /var/folders/2k/04zph6g51kj31d1hvtdwrjxw0000gn/T/d20150617-25012-1ps7wj6 --single-branch --depth 1 --branch swift-2.0
Cloning into '/var/folders/2k/04zph6g51kj31d1hvtdwrjxw0000gn/T/d20150617-25012-1ps7wj6'...
Permission denied (publickey).
fatal: Could not read from remote repository.
Please make sure you have the correct access rights and the repository exists.
```

**Comment by segiddins** [comment]
@anlaital that's an issue with git not being setup with your ssh keys (since you're using a git rather than http a URL), not with CocoaPods.

**Comment by anlaital** [comment]
@segiddins Yeah, noticed just now that it seems that all my GitHub SSH keys have died. Sorry about that. :)

**Comment by schickling** [comment]
+1

**Comment by segiddins** [comment]
Just confirming that this also works with Xcode 7 beta 2!

**Comment by segiddins** [comment]
(We'll eventually need to update the doc comments to the new format)

**Comment by jpsim** [comment]
We've posted installation instructions for using Swift 2 in your project here: https://realm.io/docs/swift/latest/#swift-20

We'll be merging this PR into `master` in the near future and releasing Swift 2 versions of Realm Swift alongside our regular releases once we have the infrastructure in place. We'll update this issue as we make progress.

**Comment by tgoyne** [comment]
Closing in favor of #2209, which makes `master` support both Swift 1.2 and Swift 2.0.

**Pull request closed without merging**